VideoApi/VideoApi.Services/AzureStorageServiceBase.cs

```diff
@@ -58,10 +66,34 @@ await foreach (var blob in allBlobs)
                     blobFullNames.Add(blob.Name);
                 }
             }
+            return blobFullNames;
+        }
+
+        public Task<IEnumerable<string>> GetAllEmptyBlobsByFilePathPrefix(string filePathPrefix)
+        {
+            var allBlobsAsync = GetAllBlobsAsync(filePathPrefix);
+            return GetAllEmptyBlobs(allBlobsAsync);
+        }
+
+        private async Task<IEnumerable<string>> GetAllEmptyBlobs(IAsyncEnumerable<BlobClient> allBlobs, string fileExtension = ".mp4")
+        {
+            var blobFullNames = new List<string>();
+
+            await foreach (var blob in allBlobs)
```

Comment:
```suggestion
            await foreach (var blob in allBlobs.Where(x => x.Name.ToLower().EndsWith(fileExtension.ToLower()))
```

And remove if statement inside

Reply:
Hi Farron, it is a asyn enumerable, so I will have to add a new package lync async for this. I will suggest we will live with it for now.